compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/InterfaceLowering.kt

```diff
@@ -96,7 +96,10 @@ internal class InterfaceLowering(val context: JvmBackendContext) : IrElementTran
                  *    ```
                  */
                 function.origin == IrDeclarationOrigin.FAKE_OVERRIDE -> {
-                    val implementation = function.resolveFakeOverride()!!
+                    // We use `resolveFakeOverrideWithBugCompatibility` due to a front-end bug (KT-36188) where there could be multiple
+                    // fake-overridden implementations of a default stub function.
+                    val implementation =
+                        function.resolveFakeOverrideWithBugCompatibility() ?: error("No implementation found for: ${function.render()}")
```

Comment:
Now that we have this workaround localized... :) Observe that a few lines below, we actually ignore functions for default parameters. So it seems that we could factor the name checking logic out of the new function and just `continue` the loop here if the fake override's name ends with `$default`.

This would make `resolveFakeOverrideWithBugCompatibility` not needed, at least for now. What do you think?

Reply:
Makes sense to me! Updated. (This PR has shrunk quite a bit from the original.) 🙂